glean-core/android/src/test/java/mozilla/telemetry/glean/GleanTest.kt

```diff
@@ -927,4 +927,30 @@ class GleanTest {
         assertTrue("Difference should be more than 0 seconds, was $diff", diff > 0)
         assertTrue("Difference should be less than 2 seconds, was $diff", diff < 2000)
     }
+
+    @Test
+    fun `test glean with custom data path is set correctly`() {
+        // Initialize with a custom data path and ensure `isCustomDataPath` is true.
+        Glean.testDestroyGleanHandle()
+        val cfg = Configuration(dataPath = "glean_test")
+        Glean.initialize(context, true, cfg, buildInfo = GleanBuildInfo.buildInfo)
+        assertTrue(Glean.isCustomDataPath)
+
+        // Initialize without a custom data path and ensure `isCustomDataPath` is false.
+        Glean.testDestroyGleanHandle()
+        Glean.initialize(context, true, buildInfo = GleanBuildInfo.buildInfo)
+        assertFalse(Glean.isCustomDataPath)
+    }
+
+    @Test
+    fun `test glean does not initialize with invalid DB path`() {
+        Glean.testDestroyGleanHandle()
+
+        // The path provided here is invalid because of the leading and trailing spaces.
+        val cfg = Configuration(dataPath = " invalid db path ")
```

Comment:
Hm, honestly I would expect a path with spaces to work just fine. What's the reason this doesn't work?

Reply:
I added checks to make sure there aren't spaces at the start or end of the string. The spaces in the words are fine, its just if they are at the start or finish. It just seemed like a easy enough thing to check that would catch some weird cases.

If this doesn't seem like something we should do, then I can change that logic or have it check for other stuff.

Reply:
Fixed in https://github.com/mozilla/glean/pull/2433/commits/f237690dd8a2518ed2758d36a69b055bc0ff92bc